host/schedule_test.go

```diff
@@ -0,0 +1,541 @@
+	}
+	s.sdkClient = sdkClient
+	s.taskQueue = s.randomizeStr("tq")
+	s.worker = worker.New(s.sdkClient, s.taskQueue, worker.Options{})
+	if err := s.worker.Start(); err != nil {
+		s.Logger.Fatal("Error when starting worker", tag.Error(err))
+	}
+}
+
+func (s *scheduleIntegrationSuite) TearDownTest() {
+	s.worker.Stop()
+	s.sdkClient.Close()
+}
+
+func (s *scheduleIntegrationSuite) TestBasics() {
+	// sqlite tests are run without elasticsearch
+	if s.esClient == nil {
+		s.T().Skip()
+	}
+
+	sid := "sched-test-basics"
+	wid := "sched-test-basics-wf"
+	wt := "sched-test-basics-wt"
+	wt2 := "sched-test-basics-wt2"
+
+	// switch this to test with search attribute mapper:
+	// csa := "AliasForCustomKeywordField"
+	csa := "CustomKeywordField"
+
+	wfMemo := payload.EncodeString("workflow memo")
+	wfSAValue := payload.EncodeString("workflow sa value")
+	schMemo := payload.EncodeString("schedule memo")
+	schSAValue := payload.EncodeString("schedule sa value")
+
+	schedule := &schedulepb.Schedule{
+		Spec: &schedulepb.ScheduleSpec{
+			Interval: []*schedulepb.IntervalSpec{
+				{Interval: timestamp.DurationPtr(3 * time.Second)},
+			},
+		},
+		Action: &schedulepb.ScheduleAction{
+			Action: &schedulepb.ScheduleAction_StartWorkflow{
+				StartWorkflow: &workflowpb.NewWorkflowExecutionInfo{
+					WorkflowId:   wid,
+					WorkflowType: &commonpb.WorkflowType{Name: wt},
+					TaskQueue:    &taskqueuepb.TaskQueue{Name: s.taskQueue},
+					Memo: &commonpb.Memo{
+						Fields: map[string]*commonpb.Payload{"wfmemo1": wfMemo},
+					},
+					SearchAttributes: &commonpb.SearchAttributes{
+						IndexedFields: map[string]*commonpb.Payload{csa: wfSAValue},
+					},
+				},
+			},
+		},
+	}
+	req := &workflowservice.CreateScheduleRequest{
+		Namespace:  s.namespace,
+		ScheduleId: sid,
+		Schedule:   schedule,
+		Identity:   "test",
+		RequestId:  uuid.New(),
+		Memo: &commonpb.Memo{
+			Fields: map[string]*commonpb.Payload{"schedmemo1": schMemo},
+		},
+		SearchAttributes: &commonpb.SearchAttributes{
+			IndexedFields: map[string]*commonpb.Payload{csa: schSAValue},
+		},
+	}
+
+	var runs, runs2 int32
+	workflowFn := func(ctx workflow.Context) error {
+		workflow.SideEffect(ctx, func(ctx workflow.Context) any {
+			atomic.AddInt32(&runs, 1)
+			return 0
+		})
+		return nil
+	}
+	s.worker.RegisterWorkflowWithOptions(workflowFn, workflow.RegisterOptions{Name: wt})
+	workflow2Fn := func(ctx workflow.Context) error {
+		workflow.SideEffect(ctx, func(ctx workflow.Context) any {
+			atomic.AddInt32(&runs2, 1)
+			return 0
+		})
+		return nil
+	}
+	s.worker.RegisterWorkflowWithOptions(workflow2Fn, workflow.RegisterOptions{Name: wt2})
+
+	// create
+
+	createTime := time.Now()
+	_, err := s.engine.CreateSchedule(NewContext(), req)
+	s.NoError(err)
+
+	// sleep until we see two runs, plus a bit more
+	s.Eventually(func() bool { return atomic.LoadInt32(&runs) == 2 }, 8*time.Second, 200*time.Millisecond)
+	time.Sleep(100 * time.Millisecond)
```

Comment:
why we need this?

Reply:
Does len(describeResp.Info.RunningWorkflows)) depend on visibility? If so, you will need to use that s.Eventually with duration set to 10s, otherwise it could become flaky.

Reply:
no, it comes from the schedule start workflow results, and then checks with history directly

Reply:
Alright. But still if you feel you need to wait, then 100ms probably will become source of flaky because it is too short.